webviz_subsurface/_datainput/parse_model_file.py

```diff
@@ -0,0 +1,133 @@
+        surface_names.append(name.get_text())
+    return surface_names
+
+
+def extract_topofzone_names(basedir):
+    model_file = os.path.join(basedir, "model_file.xml")
+    with open(model_file, "r") as file:
+        soup = BeautifulSoup(file, "xml")
+    surface_wrappers = soup.find_all("surface")
+    topofzone_names = []
+    for element in surface_wrappers:
+        name = element.find("top-of-zone")
+        topofzone_names.append(name.get_text())
+    return topofzone_names
+
+
+def get_surface_files(basedir):
+    surface_names = extract_surface_names(basedir)
+    surface_dir = os.path.join(basedir, "output", "surfaces")
+    surface_files = [
+        os.path.join(surface_dir, "d_" + s + ".rxb") for s in surface_names
+    ]
+    for path in surface_files:
+        if not os.path.isfile(path):
+            raise FileNotFoundError
+    return surface_files
+
+
+def get_surface_de_files(basedir):
+    surface_names = extract_surface_names(basedir)
+    surface_dir = os.path.join(basedir, "output", "surfaces")
+    surface_de_files = [
+        os.path.join(surface_dir, "de_" + s + ".rxb") for s in surface_names
+    ]
+    for path in surface_de_files:
+        if not os.path.isfile(path):
+            return None
+    return surface_de_files
+
+
+def get_surface_dr_files(basedir):
+    surface_names = extract_surface_names(basedir)
+    surface_dir = os.path.join(basedir, "output", "surfaces")
+    surface_dr_files = [
+        os.path.join(surface_dir, "dr_" + surface_name + ".rxb")
+        for surface_name in surface_names
+    ]
+    for path in surface_dr_files:
+        if not os.path.isfile(path):
+            return None
+    return surface_dr_files
+
+
+def get_surface_dre_files(basedir):
+    surface_names = extract_surface_names(basedir)
+    surface_dir = os.path.join(basedir, "output", "surfaces")
+    surface_dre_files = [
+        os.path.join(surface_dir, "dre_" + surface_name + ".rxb")
+        for surface_name in surface_names
+    ]
+    for path in surface_dre_files:
+        if not os.path.isfile(path):
+            return None
+    return surface_dre_files
+
+
+def get_surface_dt_files(basedir):
+    surface_names = extract_surface_names(basedir)
+    surface_dir = os.path.join(basedir, "output", "surfaces")
+    surface_dt_files = [
+        os.path.join(surface_dir, "dt_" + surface_name + ".rxb")
+        for surface_name in surface_names
+    ]
+    for path in surface_dt_files:
+        if not os.path.isfile(path):
+            return None
+    return surface_dt_files
+
+
+def get_surface_dte_files(basedir):
+    surface_names = extract_surface_names(basedir)
+    surface_dir = os.path.join(basedir, "output", "surfaces")
+    surface_dte_files = [
+        os.path.join(surface_dir, "dte_" + surface_name + ".rxb")
+        for surface_name in surface_names
+    ]
+    for path in surface_dte_files:
+        if not os.path.isfile(path):
+            return None
+    return surface_dte_files
```

Comment:
The functions above defined after each other are quite similar. Turn them into one with some additional parameters (in order to deal with the small differences)?